chain/chain/src/store.rs

```diff
@@ -1853,6 +1880,27 @@ impl<'a> ChainStoreUpdate<'a> {
         self.trie_changes.push(trie_changes);
     }
 
+    pub fn add_state_changes_for_split_states(
+        &mut self,
+        block_hash: CryptoHash,
+        shard_id: ShardId,
+        state_changes: StateChangesForSplitStates,
+    ) {
+        let prev =
+            self.add_state_changes_for_split_states.insert((block_hash, shard_id), state_changes);
+        // We should not save state changes for the same chunk twice
+        assert!(prev.is_none());
+    }
+
+    pub fn remove_state_changes_for_split_states(
+        &mut self,
+        block_hash: CryptoHash,
+        shard_id: ShardId,
+    ) {
+        // We should not remove state changes for the same chunk twice
+        assert!(self.remove_state_changes_for_split_states.insert((block_hash, shard_id)));
```

Comment:
The style here is not consistent with the function above :) I personally prefer assertions to not have side effects

Reply:
Updated

Reply:
+1 while `assert!` is guaranteed to be always on in Rust, and it's OK to, eg, rely on that in unsafe code, splitting side-effectful part of it into a separate statement makes the code more obviously clear.